ProjectBasic/app/src/main/java/com/example/moviedb/screen/MainViewPageAdapter.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.moviedb.screen
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentStatePagerAdapter
+import com.example.moviedb.screen.favorite.FavoriteFragment
+import com.example.moviedb.screen.home.HomeFragment
+
+class MainViewPageAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {
+
+    override fun getItem(postion: Int): Fragment {
+        return when (postion) {
+            MainPage.HOME_PAGE-> HomeFragment()
+            MainPage.FAVORITE_PAGE -> FavoriteFragment()
```

Comment:
Suggest:
https://medium.com/@azjkjensen/using-the-newinstance-pattern-in-kotlin-e40c1b4ba1ef
https://stackoverflow.com/questions/47104345/kotlin-idiomatic-way-to-create-a-fragment-newinstance-pattern
```suggestion
            MainPage.HOME_PAGE-> HomeFragment. newInstance()
            MainPage.FAVORITE_PAGE -> FavoriteFragment.newInstance()
```